src/components/bmfont-text.js

```diff
@@ -0,0 +1,333 @@
+var path = require('path');
+var assign = require('object-assign');
+var createSDF = require('three-bmfont-text/shaders/sdf');
+var createMSDF = require('three-bmfont-text/shaders/msdf');
+var createBasic = require('three-bmfont-text/shaders/basic');
+
+var coreShader = require('../core/shader');
+var shaders = coreShader.shaders;
+var shaderNames = coreShader.shaderNames;
+
+var alignments = ['left', 'right', 'center'];
+var anchors = ['left', 'right', 'center', 'align'];
+var baselines = ['top', 'center', 'bottom'];
+
+var DEFAULT_WIDTH = 1; // 1 matches other AFRAME default widths... 5 matches prior bmfont examples etc.
+
+// @bryik set anisotropy to 16 because I think it improves the look
+// of large amounts of text particularly when viewed from an angle.
+var MAX_ANISOTROPY = 16;
+
+var FONT_BASE_URL = 'https://cdn.aframe.io/fonts/';
+var fontMap = {
+  'default': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'Aileron-Semibold': FONT_BASE_URL + 'Aileron-Semibold.fnt',
+  'DejaVu': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'Exo2Bold': FONT_BASE_URL + 'Exo2Bold.fnt',
+  'Exo2SemiBold': FONT_BASE_URL + 'Exo2SemiBold.fnt',
+  'KelsonSans': FONT_BASE_URL + 'KelsonSans.fnt',
+  'Monoid': FONT_BASE_URL + 'Monoid.fnt',
+  'SourceCodePro': FONT_BASE_URL + 'SourceCodePro.fnt',
+  'mozillavr': FONT_BASE_URL + 'mozillavr.fnt'
+};
+
+var loadedFontPromises = {};
+var loadedTexturePromises = {};
+
+coreShader.registerShader('modified-SDF', {
+  schema: {
+    map: { type: 'map', is: 'uniform' },
+    color: { type: 'color', is: 'uniform', default: 'white' },
+    opacity: { type: 'number', is: 'uniform', default: 1.0 }
+  },
+  vertexShader: [
+    'varying vec2 vUV;',
+    'void main(void) {',
+    '  gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);',
+    '  vUV = uv;',
+    '}'
+  ].join('\n'),
+  fragmentShader: [
+    'uniform sampler2D map;',
+    'uniform vec3 color;',
+    'uniform float opacity;',
+    'varying vec2 vUV;',
+    '#define ALL_SMOOTH 0.5',
+    '#define ALL_ROUGH 0.25',
+    'float aastep(float value) {',
+    '  float afwidth = (1.0 / 32.0) * (1.4142135623730951 / (2.0 * gl_FragCoord.w));',
+    '  float smooth = smoothstep(0.5 - afwidth, 0.5 + afwidth, value);',
+    '  float ratio = (gl_FragCoord.w >= ALL_SMOOTH) ? 1.0 : (gl_FragCoord.w < ALL_ROUGH) ? 0.0 : (gl_FragCoord.w - ALL_ROUGH) / (ALL_SMOOTH - ALL_ROUGH);',
+    '  return smooth * ratio + (1.0 - ratio) * value;',
+    '}',
+    'void main() {',
+    '  vec4 texColor = texture2D(map, vUV);',
+    '  float alpha = aastep(texColor.a);',
+    '  gl_FragColor = vec4(color, opacity * alpha);',
+    '}'
+  ].join('\n')
+});
+
+module.exports.Component = registerComponent('bmfont-text', {
+  schema: {
+    // scale is now determined by width and wrappixels/wrapcount... scale: {default: 0.003},
+    tabSize: {default: 4},
+    anchor: {default: 'center', oneOf: anchors}, // center default to match primitives like plane; if 'align', null or undefined, same as align
```

Comment:
Can we have all prop comments on top of the props with cap + period. there's also some double inline comments below.

Alternatively, you can add them to a docstring on top of the component (or just in the documentation)

```
**
* Text component.
*
* @param {number} width - Use A-Frame units (meters). Defaults to geometry width or DEFAULT_WIDTH.
*/
```